File: php/src/Framework/SecurityUser/SqliteSecurityUserRepository.php
<?php
declare(strict_types=1);

namespace Demo\App\Framework\SecurityUser;

use Demo\App\Framework\Database\DatabaseConnection;
use Demo\App\Framework\Database\SqliteConnection;

final class SqliteSecurityUserRepository implements SecurityUserRepository
{
    private DatabaseConnection $dbConnection;
    public function __construct(SqliteConnection $connection)
    {
        $this->dbConnection = $connection;
    }
    
    public function findUserById(string $id): ?SecurityUser
    {
        $result = $this->dbConnection->query(sprintf('SELECT * FROM users WHERE id = \'%s\'', $id));
        if(!$result) {
            return null;
        }

        $row = $result[0];

        return new SecurityUser(
            $row['id'], $row['email'], $row['password'], $row['role'], $row['status']
        );
    }
}

File: kotlin/src/main/kotlin/framework/securityuser/SqliteSecurityUserRepository.kt
package framework.securityuser

import framework.database.DatabaseConnection

class SqliteSecurityUserRepository(private val connection: DatabaseConnection) : SecurityUserRepository {

    override fun findUserById(id: String): SecurityUser? {
        val result = connection.query("SELECT * FROM users WHERE id = '$id'")
        if (!result.next()) {
            return null
        }

        return SecurityUser(
            result.getString("id"),
            result.getString("email"),
            result.getString("password"),
            result.getString("role"),
            result.getString("status"),
        )
    }
}
